### PR TITLE
[ Quorum ] Added fix for raft timestamp issue

### PR DESCRIPTION
**Pull request by srinivasansankaran**
Signed-off-by: srinivasan.sankaran <srinivasan.sankaran@accenture.com>

**Changelog**
- Added protocol in expressapp-quorum/templates/deployment.yaml file as env variable
- Added protocol in expressapi-quorum.tpl to make it available in value files
- Added env protocol in quorum dockerfile
- Added env variable in quorum config.js file
- Added code to do timestamp conversion based on consensus used in the container.js,product.js and general.js routes
- Added code to fetch protocol as variable to make it available to routes in examples/supplychain-app/quorum/express_nodeJS/web3services.js

 

**Reviewed by**
@suvajit-sarkar 
@lakshyakumar 

 

**Linked issue**
#581

### CODE REVIEW

File: examples/supplychain-app/quorum/express_nodeJS/controllers/container.js [hunk 1]
```diff
@@ -15,6 +16,7 @@ router.get("/:trackingID?", function(req, res) {
     // TODO: Implement getContainerByID functionality
     const trackingID = req.params.trackingID;
     console.log(trackingID, "***");
+    console.log(Protocol,"*******");
```
Comment (lakshyakumar) [inline examples/supplychain-app/quorum/express_nodeJS/controllers/container.js:19]: is the `console.log` required?

File: examples/supplychain-app/quorum/express_nodeJS/controllers/container.js [hunk 2]
```diff
@@ -1,11 +1,12 @@
 var express = require('express')
   , router = express.Router();
 
-const {productContract, fromAddress, fromNodeSubject} = require('../web3services');
+const {productContract, fromAddress, fromNodeSubject,Protocol} = require('../web3services');
 var multer = require('multer'); // v1.0.5
 var upload = multer(); // for parsing multipart/form-data
 var bodyParser = require('body-parser');
 
+
```
Comment (lakshyakumar) [inline examples/supplychain-app/quorum/express_nodeJS/controllers/container.js:9]: remove extra lines

File: examples/supplychain-app/quorum/express_nodeJS/web3services.js [hunk 3]
```diff
@@ -16,6 +16,9 @@ const fromNodeSubject = nodeSubject;
 console.log(fromNodeSubject);
 console.log(fromAddress+","+fromNodeSubject);
 
+const Protocol = protocol;
+console.log(Protocol);
```
Comment (lakshyakumar) [inline examples/supplychain-app/quorum/express_nodeJS/web3services.js:20]: is `console.log()` required?

**Review by lakshyakumar — CHANGES_REQUESTED** [review]

File: examples/supplychain-app/quorum/express_nodeJS/controllers/container.js (hunk 1 above)
Comment (srinivasansankaran) [reply examples/supplychain-app/quorum/express_nodeJS/controllers/container.js:19]: Would be good to have it.. can remove incase harmful

File: examples/supplychain-app/quorum/express_nodeJS/controllers/container.js (hunk 2 above)
Comment (srinivasansankaran) [reply examples/supplychain-app/quorum/express_nodeJS/controllers/container.js:9]: Sure

File: examples/supplychain-app/quorum/express_nodeJS/web3services.js (hunk 3 above)
Comment (srinivasansankaran) [reply examples/supplychain-app/quorum/express_nodeJS/web3services.js:20]: Yes this is required.

**Review by suvajit-sarkar — APPROVED** [review]
LGTM

**Review by suvajit-sarkar — CHANGES_REQUESTED** [review]
Please make the case of protocol variable consistent (lowercase preferable)

**Review by sownak — APPROVED** [review]
LGTM if all comments by Lakshya is addressed

**Review by lakshyakumar — APPROVED** [review]
LGTM

**Review by suvajit-sarkar — APPROVED** [review]

**Pull request merged**